pkg/actions/fargate/fargate.go

```diff
@@ -2,17 +2,23 @@ package fargate
 
 import (
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/cfn/manager"
 	"github.com/weaveworks/eksctl/pkg/eks"
+	"k8s.io/client-go/kubernetes"
 )
 
 type Manager struct {
-	ctl *eks.ClusterProvider
-	cfg *api.ClusterConfig
+	ctl             *eks.ClusterProvider
+	cfg             *api.ClusterConfig
+	stackManager    manager.StackManager
+	newStdClientSet func() (kubernetes.Interface, error)
 }
 
-func New(cfg *api.ClusterConfig, ctl *eks.ClusterProvider) *Manager {
+func New(cfg *api.ClusterConfig, ctl *eks.ClusterProvider, stackManager manager.StackManager) *Manager {
 	return &Manager{
-		ctl: ctl,
-		cfg: cfg,
+		ctl:             ctl,
+		cfg:             cfg,
+		stackManager:    stackManager,
+		newStdClientSet: func() (kubernetes.Interface, error) { return ctl.NewStdClientSet(cfg) },
```

Comment:
I really dislike this, but we need a way to inject the clientset creator, which is currently on the provider and not mockable :cry: